.github/workflows/build.yml

```diff
@@ -202,8 +202,8 @@ jobs:
           exit 1
         fi
 
-        # Check that commit contains ':'
-        if commit_titles |  grep -E -v '^([.a-z]+, )*[.a-z]+: [a-z]'; then
+        # Check that commit contains ':' (and no '!')
+        if commit_titles | grep -E -v '^([.a-z\-]+, )*[.a-z\-]+: [a-z\-]'; then
```

Comment:
- The title should be in lowercase
- We can separate multiple components using `:` in case of hierarchical relationships

```suggestion
        if commit_titles | grep -E -v '^([.a-z\-]+[,:] )*[.a-z\-]+: [a-z][a-z0-9 _.,\-]*$'; then
```